api/rbac/rbac.proto

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2020-present Open Networking Foundation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+package onos.lib.go.rbac;
+
+
+service Rbac {
+    // TODO add required RPC functions for creating, updating, deleting, and retrieving the roles in the system
```

Comment:
I understand your point about a circular logic. I think there is a tradeoff here and we can think about it in two different ways and perhaps a discussion will be needed to make a decision about it. 
1- One approach is to have a centralized server which is responsible to provide a programming API that allows us to create/update/delete/retrieve custom roles and store them in a store (e.g. a map). From the interceptor point of view, it uses this service just to retrieve  the current roles in the system (instead of reading it from a file) to authorize a gRPC call and I don't think retrieving the roles via a remote server has any conflict with authorization. This approach is good if we want to have a user interface (via gui for management platform) that allows us to create new roles in the system without touching the running systems or list them. For example, suppose we have a new group joined and we would like to give them access to a set of services in the cluster, we just need to add a new role to update role store that can be used later without changing the config file which should be defined part of the helm chart. Perhaps the name that I gave to the service is misleading (it should be role service or role server or something like that) because that service just provides a way of accessing the roles in the system and doesn't do anything in terms of authorization.  This is the approach which is also used in the KubeCon example (https://github.com/libopenstorage/openstorage/blob/5968638c45e46c6bc878fdf95387ada863f1d94e/api/api.proto#L1304). 

2- If my understanding is correct based on what you said, the second approach is to define roles per subsystem and keep it local as you suggested via a YAML file. In my opinion, some of the roles potentially can be duplicated. For example, an admin role should have access to all of the services and apis and we just need one role to define it instead of duplicating it for each subsystem.

Reply:
This service is not necessary needed to be deployed per subsystem, however it should be a standalone service which exposes a port that can be used by other services which use Authorization interceptor.